### PR TITLE
[LLSC-20] Implement Skeleton Email Service

### PR DESCRIPTION
**Pull request by Mayank808**
## Notion ticket link

[Setup Basic Email System](https://www.notion.so/uwblueprintexecs/Setup-Basic-Email-System-11110f3fb1dc80a48e88ce63228b926c?pvs=4)

## Implementation description
Setup a quick skeleton for email service with interfaces and classes defined. Will be implementing actual functionality in next pr. Created a test endpoint to ensure that implementation of skeleton classes is done properly just need to add actual logic.

## Steps to test
1. Can run backend properly 
2. Hit the new test email API. Can check fast API docs for params/requirements

## What should reviewers focus on?
* Interface definitions other functionalities that we may ned

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/app/services/email/email_service_provider.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from app.services.interfaces.email_service_provider import IEmailServiceProvider
```
Comment (Mayank808) [inline backend/app/services/email/email_service_provider.py:1]: i think we should move the interfaces folder directly under the app directory. Let me know if yall agree.

File: backend/app/routes/email.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+from app.services.email.email_service_provider import AmazonSESEmailProvider
+
+router = APIRouter()
+
+def get_email_service() -> IEmailService:
+    email_provider = AmazonSESEmailProvider(aws_access_key = "", aws_secret_key ="")
+    return EmailService(email_provider)
+
+@router.post("/send-test-email/")
+async def send_welcome_email(recipient: str, user_name: str, email_service: IEmailService = Depends(get_email_service)):
```
Comment (mslwang) [inline backend/app/routes/email.py:13]:
```suggestion
async def send_welcome_email(recipient: str, user_name: str, email_service: Annotated[IEmailService, Depends(get_email_service)]):
``` 

I think it's recommended to use [Annotated](https://fastapi.tiangolo.com/tutorial/dependencies/#share-annotated-dependencies), at least for code deduplication it makes sense. Don't need to spend too much time if it doesn't work though.

File: backend/app/services/interfaces/email_service_provider.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+from abc import ABC, abstractmethod
+
+class IEmailServiceProvider(ABC):
```
Comment (mslwang) [inline backend/app/services/interfaces/email_service_provider.py:3]: Thanks for adding this, I think this is a good idea if we decide to swap out Email providers or decide to mock the provider.

File: backend/app/services/email/email_service.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    def send_welcome_email(self, recipient: str, user_name: str) -> dict:
+        pass
+
+    def send_password_reset_email(self, recipient: str, reset_link: str) -> dict:
+        pass
+
+    def send_notification_email(self, recipient: str, notification_text: str) -> dict:
+        pass
+    
+    def _helper_method_example(self):
```
[the remark this answers is not in the document]
Comment (Mayank808) [reply backend/app/services/email/email_service.py:20]: i just put it as an example for how those should be formated ie with a _ at the front but yeah ill get ride of it

**Review by mslwang — APPROVED** [review]
Looks good, but let's make this a separate ticket from the one specified, as the one provided includes the requirement to use the email service provider itself. We can make using the third party email service provider its own ticket.

**Review by EdmondLi1 — APPROVED** [review]
LGTM

**Pull request merged**